Typescript 4.7+ `NodeNext/Node16` incompatibility

TypeScript 4.7 [added full support for ESM Modules](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#ecmascript-module-support-in-node-js) via two new tsconfig `module` settings - `nodenext` and `node16`.

One of the caveats of these new module resolution modes is that:

> Relative import paths need full extensions (we have to write import "./foo.js" instead of import "./foo").

Unfortunately the TypeScript code is missing `.js` import extensions throughout.

Does this issue explain why `Konva.Stage` has type `any` and why I can’t use `react-konva`?

```
error TS2322: Type '{ width: number; }' is not assignable to type 'IntrinsicAttributes & StageProps & KonvaNodeEvents & ClassAttributes<any>'.
  Property 'width' does not exist on type 'IntrinsicAttributes & StageProps & KonvaNodeEvents & ClassAttributes<any>'.

130       <Stage width={300}></Stage>
                 ~~~~~
```

If yes, what can I do to work around this issue?

--- a/rename-imports.mjs
+++ b/rename-imports.mjs
@@ -1,7 +1,7 @@
 import FileHound from 'filehound';
 import fs from 'fs';
 
-const files = FileHound.create().paths('./lib').ext('js').find();
+const files = FileHound.create().paths('./lib').ext(['js', 'ts']).find();
 
 files.then((filePaths) => {
   filePaths.forEach((filepath) => {
@@ -21,6 +21,9 @@ files.then((filePaths) => {
         "import * as Canvas from 'canvas.js';",
         "import * as Canvas from 'canvas';"
       );
+
+      // Handle import("./x/y/z") syntax.
+      text = text.replace(/(import\s*\(\s*['"])(.*)(?=['"])/g, '$1$2.js');
 
       fs.writeFile(filepath, text, function (err) {
         if (err) {

